### PR TITLE
python3Packages.lizard: init at 1.17.10

### PR DESCRIPTION
**Pull request by jpetrucciani**
###### Description of changes

I would like to be able to use the `lizard` ([link here](https://github.com/terryyin/lizard)) code metrics tool inside of nixpkgs!

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/lizard/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  pname = "lizard";
+  version = "1.17.10";
+  disabled = pythonOlder "3.6";
```
Comment (fabaff) [inline pkgs/development/python-modules/lizard/default.nix:10]: I would suggest to limit it to Python releases which are not EOL. 

```suggestion
  disabled = pythonOlder "3.7";
```

File: pkgs/development/python-modules/lizard/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1p2s0wx1l75dsljiyvars7xlljzs1mipm8jaby5y4jvjck6qmmv2";
+  };
+
+  pythonImportsCheck = [
+    "lizard"
+  ];
+
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/development/python-modules/lizard/default.nix:21]: The repo seems to contain [tests](https://github.com/terryyin/lizard/tree/master/test).

File: pkgs/development/python-modules/lizard/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  meta = with lib; {
+    description =
+      "A code analyzer without caring the C/C++ header files. It works with Java, C/C++, JavaScript, Python, Ruby, Swift, Objective C. Metrics includes cyclomatic complexity number etc";
```
Comment (fabaff) [inline pkgs/development/python-modules/lizard/default.nix:25]:
```suggestion
    description =
      "Code analyzer without caring the C/C++ header files";
```

The extended content could go into `longDescription`.

File: pkgs/development/python-modules/lizard/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    homepage = "http://www.lizard.ws";
```
Comment (fabaff) [inline pkgs/development/python-modules/lizard/default.nix:26]: The license needs to be set.

```suggestion
    homepage = "http://www.lizard.ws";
    license = with licenses; [ mit ];
```

File: pkgs/development/python-modules/lizard/default.nix (hunk 1 above)
Comment (jpetrucciani) [reply pkgs/development/python-modules/lizard/default.nix:10]: Great idea! I'll update

File: pkgs/development/python-modules/lizard/default.nix (hunk 3 above)
Comment (jpetrucciani) [reply pkgs/development/python-modules/lizard/default.nix:25]: makes sense! I'll change it

File: pkgs/development/python-modules/lizard/default.nix (hunk 4 above)
Comment (jpetrucciani) [reply pkgs/development/python-modules/lizard/default.nix:26]: good catch! I'll add it

File: pkgs/development/python-modules/lizard/default.nix (hunk 2 above)
Comment (jpetrucciani) [reply pkgs/development/python-modules/lizard/default.nix]: good call - another one where i spaced on pulling from github instead of pypi 🤦‍♂️

**Comment by jpetrucciani** [comment]
@fabaff Thanks for the review here too! 😄

File: pkgs/development/python-modules/lizard/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    rev = version;
+    sha256 = "sha256-4jq6gXpI1hFtX7ka2c/qQ+S6vZCThKOGhQwJ2FOYItY=";
+  };
+
+  checkInputs = [
+    pytestCheckHook
+    mock
+  ];
+
+  propagatedBuildInputs = [ jinja2 ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/lizard/default.nix:27]:
```suggestion
  propagatedBuildInputs = [ jinja2 ];

  checkInputs = [
    pytestCheckHook
    mock
  ];
```

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.lizard

**Pull request merged**